### DISCUSSION TITLE
Fonts passed to `/forms/chromium/convert/html` do not render if `pdfa=PDF/A-2b`

### DISCUSSION THREAD

**Discussion by Vovcharaa** [gotenberg/gotenberg#986] [Q&A]
PDF/A-2 specifies that it can embed OpenType fonts.
But when I try to do `src: url('futura-bold.otf') format('opentype');` in css and set `pdfa=PDF/A-2b` font is not used in pdf.
How to use custom fonts with PDF/A-2b/3b?

**Comment by gulien** [comment]
Hello @Vovcharaa 👋 

That’s because the resulting PDF from Chromium is post-processed by LibreOffice, and it may remove custom fonts from it. A solution could be to install the font directly in the Docker image: https://gotenberg.dev/docs/configuration. Not sure it would work though.

**Reply by Vovcharaa** [reply]
I tried to mount the same font I used to render pdf, but there was no effect.
Perhaps, I need to try using mounted font already in Chrome without `src` in CSS

**Reply by Vovcharaa** [reply]
https://help.libreoffice.org/latest/en-US/text/shared/01/ref_pdf_export_general.html?&DbPAR=SHARED&System=MAC#:~:text=Archival%20(PDF/A,long%20term%20preservation.
"All fonts used in the source document are embedded in the generated PDF file."
Why is it not working in gotenberg?

**Reply by gulien** [reply]
I don’t know 🤷‍♂️ « source document » probably mention a non-PDF document. 

You may try to convert the resulting PDF from Chromium to PDF/A with your local installed LibreOffice, but chances are the resulting PDF will be the same.

**Reply by Vovcharaa** [reply]
Locally installed LibreOffice can't even use embedded font to view the file, because it opens pdf for edit.
https://ask.libreoffice.org/t/libreoffice-draw-not-displaying-fonts-properly-on-ubuntu-22-04-xfce/94941#:~:text=LibreOffice%20cannot%20use,a%20substitution%20happens.

**Reply by Vovcharaa** [reply]
Have you considered using Ghostscript for pdf to pdf/a conversion?

**Reply by gulien** [reply]
Not possible due to licenses incompatibility alas.

**Reply by gulien** [reply]
Someone did build a version with a Ghostscript module: https://github.com/Vrex123/gotenberg-ghostscript

**Reply by Vovcharaa** [reply]
I doubt that AGPL license allows me to use Ghostscript in commercial project.

**Reply by gulien** [reply]
It depends, if you pay for a license I guess it's ok 😄

**Comment by madduck** [comment]
Why is Libreoffice used to post-process the PDF with `pdfa=PDF/A-2b`, but not if I don't specify this configuration in the HTML POST?

My issue is that Libreoffice garbles ligatures. If I process an HTML file like this:

```
curl -v http://192.168.235.10:3000/forms/chromium/convert/html -F files=@/tmp/madduck/index.html -o /tmp/test.pdf
```

it generates a beautiful PDF. If I add `-F pdfa=PDF/A-2b`, then the result is ugly, with all ligatures simply dropped:

![image](https://github.com/user-attachments/assets/01027d2c-7c49-4c75-949a-3b3727e7ecc9)

**Reply by gulien** [reply]
Hello @Vovcharaa 

Because it’s the only tool with a correct license I’m aware of that can convert PDFs to PDF/A.

**Reply by madduck** [reply]
> Because it’s the only tool with a correct license I’m aware of that can convert PDFs to PDF/A.

Okay, so I am trying to reproduce the issue with Libreoffice directly in order to chase down this bug. How does the pipeline work within Gotenberg? I am not too good with Go and cannot figure it out myself. Does Gotenberg use Chrome to make a PDF, and then pass that PDF to Libreoffice?

Because if I do that, i.e. create a PDF/not-A "normal" PDF with Gotenberg and then try to convert it, I get a silly error:

```
% libreoffice --headless --convert-to pdf --outdir /tmp /tmp/madduck/test.pdf
Warning: failed to launch javaldx - java may not function correctly
Error: source file could not be loaded
```

**Comment by madduck** [comment]
Note that I can reproduce the problem sending a PDF to the `/forms/libreoffice/convert`:

```
curl -v http://192.168.235.10:3000/forms/libreoffice/convert -F files=@/tmp/test.pdf -F pdfa=PDF/A-2b -o /tmp/broken.pdf
```

this results in broken ligatures.

Can you help me find out what Gotenberg actually does within, so that I can try to reproduce this with LO directly, and report a bug there?
